File: 프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.java
class Solution {
    public int solution(int n) {
        
        return Math.sqrt(n) % 1 == 0 ? 1 : 2;
    }
}

// 다른 풀이
if (n % Math.sqrt(n) == 0) {
    return 1;
} else {
    return 2;
}

// 다른 풀이

File: 프로그래머스/0/120909. 제곱수 판별하기/제곱수 판별하기.py
import math

def solution(n):    
    
    return 1 if math.sqrt(n) % 1 == 0 else 2

# 다른 풀이
return 1 if (n ** 0.5).is_integer() else 2
# .is_integer() : 정수면 true, 아니면 false

# 다른 풀이
if n**(1/2) == int(n**(1/2)) :
    return 1
else :
    return 2
